### PR TITLE
HPCC-26361 Prometheus metrics sink service bind all NICs

### PR DESCRIPTION
**Pull request by rpastrana**
- Service should bing to all nics

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by rpastrana** [comment]
@kenrowland please review

File: system/metrics/sinks/prometheus/prometheusSink.cpp [hunk 1]
```diff
@@ -144,5 +144,5 @@ void PrometheusMetricSink::stopCollection()
 void PrometheusMetricSink::startServer()
 {
     LOG(MCoperatorProgress, "PrometheusMetricsService started:  port: '%i' uri: '%s' sinkname: '%s'\n", m_port, m_metricsServiceName.str(), m_metricsSinkName.str());
-    m_server.listen("localhost", m_port);
+    m_server.listen(BIND_ALL_LOCAL_NICS, m_port);
```
Comment (kenrowland) [inline system/metrics/sinks/prometheus/prometheusSink.cpp:147]: My only concern here is the principal of least privilege and that only listening on the IP address that the Prometheus server knows would be best. If there are no security concerns related to that, let me know and I'll approve.

**Review by kenrowland — CHANGES_REQUESTED** [review]
@rpastrana Please see question

**Comment by rpastrana** [comment]
@kenrowland good question. Prometheus  wouldn't be able to connect to "localhost" unless it were running on the same pod, and even then it wouldn't be able to connect via any external (to the machine) addresses. This allows the server socket to accept traffic from other pods (internal/external cluster traffic is controlled at the docker/k8s/helm level). As far as I know, this is accepted way to expose http services.

**Review by kenrowland — APPROVED** [review]
@rpastrana Just wanted to make sure we are OK with listening on all IP addresses. Approved.

**Comment by rpastrana** [comment]
@kenrowland agreed, it's a fair question to ask.

**Comment by rpastrana** [comment]
@richardkchapman please merge.

**Pull request merged**